### ISSUE TITLE
Uncaught TypeError: gallery.justifiedGallery is not a function

### ISSUE THREAD

**Issue by vtomyev** [mfg92/hugo-shortcode-gallery#66]
I installed the shortcode using hugo modules. When I attempt to run the shortcode nothing shows up on the screen. To be fair im a bit new to this.<br>
Here is the error I see in the console 
![Screenshot from 2023-12-31 20-22-49](https://github.com/mfg92/hugo-shortcode-gallery/assets/18566857/af3b8982-20fc-4ccb-a864-d8ff796c0ab1) <br>
This is he HTML that loads. I can see the picture there. 
![Screenshot from 2023-12-31 20-23-21](https://github.com/mfg92/hugo-shortcode-gallery/assets/18566857/a7b50add-ece4-443b-a852-8dca17eae7a1)<br>
The gallery is supposed to be between the heading and the social media buttons.
![Screenshot from 2023-12-31 20-23-36](https://github.com/mfg92/hugo-shortcode-gallery/assets/18566857/b03d4ed5-353b-4cec-aee4-e7b764f68d8f)

Can anyone please help me?

**Comment by mfg92** [comment]
Can you show how you included the gallery into your markdown file?
Also, please check your browser's devtools network tab to see if all files are loading successfully. The error you are experiencing may indicate that the Justified-Gallery JS could not be loaded.

**Comment by vtomyev** [comment]
Thank you for taking the time out of your day to help me with this!
Here is how I'm using the shortcode in my markdown: 
![image](https://github.com/mfg92/hugo-shortcode-gallery/assets/18566857/baaddee1-3ede-4c89-90c6-9fc1abc91623)
<br>
This is my network tab in the browser when I try to load the gallery page:
![image](https://github.com/mfg92/hugo-shortcode-gallery/assets/18566857/e4d807f0-27b5-46fc-b8bd-32e40268d68d)

**Comment by mfg92** [comment]
There are two versions of jQuery loaded on your page, one from hugo-shortcode-gallery (_jquery-3.7.1.min.js_) and another, probably from your theme (_jquery.min.js_). Having two versions of jQuery in parallel will lead to errors. You can tell hugo-shortcode-gallery not to load it's own jQuery, but to use the jQuery provided by your theme. To do this, simply remove the _loadJQuery_ parameter from the shortcode.

**Comment by rhizoet** [comment]
I have the same problem. I'm using the [universal theme](https://github.com/devcows/hugo-universal-theme) an the latest hugo version.

All resources load correctly (take a look at screenshot) and only in Console I see

```
Uncaught TypeError: gallery.justifiedGallery is not a function
    at HTMLDocument.<anonymous> (impressionen/:2748:11)
    at e (jquery-3.7.1.min.js:2:27028)
    at t (jquery-3.7.1.min.js:2:27330)
```

I don't know what to do. Do you have an hint for me?

![Bildschirmfoto_2024-01-21_18-29-34](https://github.com/mfg92/hugo-shortcode-gallery/assets/58997736/51f9ee78-bdcd-43b7-9db2-a044327359f5)

**Comment by mfg92** [comment]
@rhizoet Have you tried removing `loadJQuery=true` from the shortcode?

**Comment by rhizoet** [comment]
Yes, I've tried. But then I get 
`Uncaught ReferenceError: jQuery is not defined`.

So I think I must enable jQuery.

**Comment by vtomyev** [comment]
Yes, I have tried that but I get the exact same error.

**Comment by mfg92** [comment]
@rhizoet & @vtomyev: 

This is most likely due to your website/theme loading a second instance of jQuery **after** the gallery has been partially initialised.

I have implemented something that should prevent the gallery from breaking in such cases, its on the _jquery-loading-robustness_ branch. Together with `loadJQuery=true` this should solve your problem. Please try it and let me know if it works.

**Comment by rhizoet** [comment]
@mfg92 Thank you very much for your hard work. It works perfectly on my sites.

**Comment by vtomyev** [comment]
Hey, I'm sorry to be bugging you with this. Mine still is not working...
Now I'm getting this error:
![image](https://github.com/mfg92/hugo-shortcode-gallery/assets/18566857/d2e46f9e-55f8-49ef-bfcc-31745824a414)

**Comment by mfg92** [comment]
> Hey, I'm sorry to be bugging you with this. Mine still is not working... Now I'm getting this error: 

Are you sure you have checked out the **jquery-loading-robustness** branch?
If you have, and you are still getting this exception, I need more information. Best would be a ZIP or github repository containing a minimal example of a complete hugo project where this problem occurs.

**Comment by mfg92** [comment]
@rhizoet I have merged the changes into main, thank you for testing.

@vtomyev I will close this issue as I have not heard from you, if you still have problems please open a new issue with a minimal sample project that shows your problem.
